opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -125,14 +125,16 @@
     private RuntimeEnvironment() {
         configuration = new Configuration();
         configLock = new ReentrantReadWriteLock();
+        authFrameworkLock = new ReentrantReadWriteLock();
         watchDog = new WatchDogService();
         lzIndexerParallelizer = LazilyInstantiate.using(() ->
                 new IndexerParallelizer(this));
         lzSearchExecutor = LazilyInstantiate.using(() -> newSearchExecutor());
     }
 
-    /** Instance of authorization framework.*/
+    /** Instance of authorization framework and its lock.*/
     private AuthorizationFramework authFramework;
+    private final ReentrantReadWriteLock authFrameworkLock;
```

Comment:
```suggestion
    private final ReadWriteLock authFrameworkLock;
```
Program to an interface paradigm.